From 1c3fcbdaa99a2859ea008061782223f4d1b40cce Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Sat, 19 Aug 2023 21:20:03 +0200
Subject: [PATCH] tests: add 'large-time' as a testable feature

This allows test cases to require this feature to run and to be used in
%if conditions.

Large here means larger than 32 bits. Ie does not suffer from y2038.

Closes #11696

Backported by: Aquila Macedo Costa <aquilamacedo@riseup.net>.

Changes:
- Refresh patch context
---
 tests/FILEFORMAT.md     | 1 +
 tests/runtests.pl       | 1 +
 tests/server/disabled.c | 3 +++
 3 files changed, 5 insertions(+)

diff --git a/tests/FILEFORMAT.md b/tests/FILEFORMAT.md
index aff0e2d..65df41c 100644
--- a/tests/FILEFORMAT.md
+++ b/tests/FILEFORMAT.md
@@ -417,6 +417,7 @@ Features testable here are:
 - `ipv6`
 - `Kerberos`
 - `large_file`
+- `large-time` (time_t is larger than 32 bit)
 - `ld_preload`
 - `libssh2`
 - `libssh`
diff --git a/tests/runtests.pl b/tests/runtests.pl
index 1510527..c4c7717 100755
--- a/tests/runtests.pl
+++ b/tests/runtests.pl
@@ -3064,6 +3064,7 @@ sub setupfeatures {
     $feature{"wakeup"} = 1;
     $feature{"headers-api"} = 1;
     $feature{"xattr"} = 1;
+    $feature{"large-time"} = 1;
     $feature{"nghttpx"} = !!$ENV{'NGHTTPX'};
     $feature{"nghttpx-h3"} = !!$nghttpx_h3;
 }
diff --git a/tests/server/disabled.c b/tests/server/disabled.c
index 7ce2903..48cf878 100644
--- a/tests/server/disabled.c
+++ b/tests/server/disabled.c
@@ -78,6 +78,9 @@ static const char *disabled[]={
 #endif
 #ifndef USE_XATTR
   "xattr",
+#endif
+#if (SIZEOF_TIME_T < 5)
+  "large-time",
 #endif
   NULL
 };
